### PR TITLE
feat(fuzz): add XSS reflection context analyzer

### PR DESCRIPTION
**Pull request by Harry-jain**
/claim #5838

## Summary

This pull request introduces a **tokenizer-based XSS reflection context analyzer** for the Nuclei fuzzing engine.

The analyzer inspects HTTP responses generated during fuzzing and determines the **HTML context in which the injected payload (canary) is reflected**. This enables more accurate XSS detection and improves payload selection for subsequent fuzzing attempts.

The implementation follows the existing Nuclei fuzz analyzer architecture and integrates cleanly into the analyzer registry.

---

## Supported Reflection Contexts

The analyzer detects the following contexts:

- `html_text`
- `html_attribute`
- `event_handler`
- `url_attribute`
- `script_executable`
- `script_data`
- `style`
- `html_comment`

If the reflection context cannot be determined, the analyzer safely returns `unknown`.

---

## Implementation Details

Key implementation points:

- Uses **`golang.org/x/net/html` tokenizer** to safely parse HTML instead of regex-based parsing.
- Extracts the injected payload marker from `options.FuzzGenerated.Value`.
- Sends the fuzzed request using the Nuclei analyzer HTTP client.
- Parses the response body and classifies the context of the reflected marker.
- Integrates with the existing fuzz analyzer registry.

This approach avoids false positives and ensures context classification aligns with real HTML parsing behavior.

---

## Summary by CodeRabbit

* **New Features**
  * Added an XSS analyzer that detects and classifies reflection contexts in HTTP responses (text, attributes, event handlers, URLs, scripts, styles, comments).

* **Tests**
  * Added comprehensive tests covering detection across varied HTML/JS contexts and edge cases.

* **Documentation**
  * Updated syntax and template docs to include the new analyzer name as a valid option.

### CODE REVIEW

**Review by Harry-jain — COMMENTED** [review]
changes

File: pkg/fuzz/analyzers/xss/context_analyzer.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+							return "script_data"
+						}
+					}
+				}
+				return "script_executable"
+			}
+
+			if tag == "style" {
+				return "style"
+			}
```
[the remark this answers is not in the document]
Comment (Harry-jain) [reply pkg/fuzz/analyzers/xss/context_analyzer.go:123]: Good catch — thank you for the detailed analysis.

You're correct that the current implementation classifies `script_executable` and `style` as soon as the start tag is encountered, without verifying that the marker actually exists inside the element content. This can indeed lead to false positives when the marker appears outside the tag.

I will update the tokenizer logic to:

1. Continue scanning the token stream after encountering `<script>` and `<style>`.
2. Inspect the element content until the corresponding closing tag.
3. Only classify the context as `script_executable`, `script_data`, or `style` if the marker is found within that element.
4. Otherwise continue scanning so reflections outside the element are correctly detected as `html_text`.

I'll push a fix shortly that implements this behavior while preserving the existing `type` attribute handling for JSON script contexts.

Thanks again for pointing this out.

**Comment by Harry-jain** [comment]
All automated review feedback has been addressed.

Changes included:
- Added guard for empty marker to prevent false positives.
- Updated script/style detection logic to ensure the marker exists inside the element before classifying the context.
- Introduced a bounded response body read to prevent unbounded memory allocation during analysis.
- Added additional regression tests for edge cases (e.g. `<script>safe</script><div>MARKER</div>`).

The analyzer now:
- Verifies marker presence before context classification
- Handles script/style blocks correctly
- Avoids potential memory exhaustion during response parsing

Ready for maintainer review. Thanks!

**Review by Harry-jain — COMMENTED** [review]
All automated review feedback has been addressed and fixes pushed.

The analyzer now includes:
- marker validation guard
- script/style context verification
- bounded response body reads
- regression tests for edge cases

Ready for maintainer review.

**Comment by Harry-jain** [comment]
hey @Mzack9999 any updates?
please let me know of any discrepancies
